lib/PlaylistSupport/PlaylistSupport.js

```diff
@@ -0,0 +1,370 @@
+			this.collection   = (this.collection   === undefined) ? null : this.collection;
+			this.autoPlayOnShuffle = (this.autoPlayOnShuffle === undefined) ? false : this.autoPlayOnShuffle;
+			this.playPrevThreshold = (this.playPrevThreshold === undefined) ? 2 : this.playPrevThreshold;
+		};
+	}),
+
+	/**
+	* @method
+	* @private
+	*/
+	dispatchEvent: kind.inherit(function (sup) {
+		return function (sEventName, oEvent, oSender) {
+			// Needed for proper onenter/onleave handling
+			if (this.strictlyInternalEvents[sEventName] && this.isInternalEvent(oEvent)) {
+				return true;
+			}
+			// FIXME: not sure why events can arrive without event objects, but we guard here for safety
+			if (oEvent && !oEvent.delegate) {
+				var handler = this._playlist_Handlers[sEventName];
+				if (handler){
+					this.cachePoint = true;
+					if(this[handler](oSender, oEvent)) {
+						return true;
+					}
+				}
+			}
+			return sup.apply(this, arguments);
+		};
+	}),
+
+	/**
+	* Return a model contains next audio information.
+	* When decide next audio, repeat and shuffle flag are referenced. 
+	* 
+	* @public
+	*/
+	getNext: function (model) {
+		var c = this.get('collection');
+		if (this.repeat == 'one' || !c)
+			return this.get('model');
+
+		var	index = c.indexOf(model);
+		if (index != -1) {
+			index = this.shuffle ? this.playOrder.indexOf(index) : index;
+			index++;
+			index = this.repeat == 'all' ? index % c.length : Math.min(index, c.length-1);
+		} else {
+			// If it fails to find current on collection, we assume model is set directly by user.
+			// We come back to collection by set first item from collection as next
+			index = 0;
+		}
+
+		return c.at(this.shuffle ? this.playOrder[index] : index);
+	},
+
+	/**
+	* Return a model contains previous audio information.
+	* When decide next audio, repeat and shuffle flag are referenced. 
+	* 
+	* @public
+	*/
+	getPrevious: function (model) {
+		var c = this.get('collection');
+		if (this.repeat == 'one' || !c)
+			return this.get('model');
+
+		var	index = c.indexOf(model);
+		if (index != -1) {
+			index = this.shuffle ? this.playOrder.indexOf(index) : index;
+			index--;
+			index = this.repeat == 'all' ? index % c.length : Math.max(index, 0);
+		} else {
+			// If it fails to find current on collection, we assume model is set directly by user.
+			// We come back to collection by set first item from collection as previous			
+			index = 0;
+		}
+
+		return c.at(this.shuffle ? this.playOrder[index] : index);
+	},
+
+	/**
+	* Add audio into playlist.
+	* 
+	* @public
+	*/
+	addAudio: function (model) {
+		var c = this.get('collection');
+		c && c.add(model);
+	},
+
+	/**
+	* @private
+	*/
+	_addAudio: function (sender, event) {
+		this.addAudio(event.model);
+	},
+
+	/**
+	* Remote audio from playlist.
+	* 
+	* @public
+	*/
+	removeAudio: function (model) {
+		var c = this.get('collection');
+		c && c.remove(model);
+	},
+
+	/**
+	* @private
+	*/
+	_removeAudio: function (sender, event) {
```

Comment:
On each of these functions where we refer to `event`, `event` should be changed to `ev` instead, because "event" is a reserlved global variable in Internet Explorer (and when we want it to work later, we won't have to change it.)